pyvista/core/utilities/geometric_objects.py

```diff
@@ -363,6 +364,353 @@ def Sphere(
     return surf
 
 
+def SolidSphere(
+    outer_radius=0.5,
+    inner_radius=0.0,
+    radius_resolution=5,
+    start_theta=0.0,
+    end_theta=360.0,
+    theta_resolution=30,
+    start_phi=0.0,
+    end_phi=180.0,
+    phi_resolution=30,
+    radius=None,
+    theta=None,
+    phi=None,
+    center=(0.0, 0.0, 0.0),
+    direction=(0.0, 0.0, 1.0),
+):
+    """Create a solid sphere.
+
+    Parameters
+    ----------
+    outer_radius : float, default: 0.5
+        Outer radius of sphere.
+
+    inner_radius : float, default: 0.0
+        Inner radius of sphere.
+
+    radius_resolution : int, default: 5
+        Number of points in radial direction.
+
+    start_theta : float, default: 0.0
+        Starting longitude angle in degrees.
+
+    end_theta : float, default: 360.0
+        Ending longitude angle in degrees.
+
+    theta_resolution : int, default: 30
+        Number of points in theta direction.
+
+    start_phi : float, default: 0.0
+        Starting latitude angle in degrees.
+
+    end_phi : float, default: 180.0
+        Ending latitude angle in degrees.
+
+    phi_resolution : int, default: 30
+        Number of points in phi direction,
+        inclusive of polar axis if applicable.
+
+    radius : sequence[float], optional
+        A monotonically increasing sequence of values specifying radial
+        points. If specified, it overrides any choices for ``outer_radius``,
+        ``inner_radius``, and ``radius_resolution``.
+
+    theta : sequence[float], optional
+        A monotonically increasing sequence of values specifying theta
+        points in degrees. If specified, it overrides any choices for
+        ``start_theta``, ``end_theta``, and ``theta_resolution``.
+
+    phi : sequence[float], optional
+        A monotonically increasing sequence of values specifying phi
+        points in degrees. If specified, it overrides any choices for
+        ``start_phi``, ``end_phi``, and ``phi_resolution``.
+
+    center : sequence[float], default: (0.0, 0.0, 0.0)
+        Center in ``[x, y, z]``.
+
+    direction : sequence[float], default: (0.0, 0.0, 1.0)
+        Direction vector in ``[x, y, z]`` pointing from ``center`` to
+        the sphere's north pole at zero degrees latitude.
```

Comment:
Am I misunderstanding the intent of this sentence, or should this be
```suggestion
        Direction vector in ``[x, y, z]`` pointing from ``center`` to
        the sphere's north pole at 90 degrees latitude.
```
?

Reply:
This follows `pyvista.Sphere` and vtk's definitions.  In fact this is a straight copy paste from the other docstring.

Reply:
The word latitude should not be used here. Latitude is measured 0 to 90 from equator to North pole and 0 to -90 from equator to South pole, but this is not how `phi` is measured. It should say (in order of my preference) `the sphere's North pole at phi = 0`  or `phi = 0 degrees` or `0 polar angle`.

Reply:
I added this to the TODO list at top.  Let's fix both the description here and in `Sphere`

Reply:
Great work here! I am a big fan of clearing up any confusion regarding how these parameters and angles are defined. I have a few comments:
- There is a great [wolfram article on spherical coordinates](https://mathworld.wolfram.com/SphericalCoordinates.html) which highlights different conventions and the sources of confusion, and mentions how theta and phi are commonly swapped in both meaning and order.
- The current docstring language for the parameters of `Sphere` was introduced in #4674
- The wording "north pole at zero degrees latitude" comes from VTK's own [`vtkSphereSource` documention](https://vtk.org/doc/nightly/html/classvtkSphereSource.html#a106b83c71c8fa0d2909ac4a714b23325), which states
    - "`SetStartPhi()`: Set the starting latitude angle (0 is at north pole)."
- I agree that this choice is confusing, and that a north pole at +90 degrees latitude makes more sense. To redefine this would require deviation from its current alignment with VTK usage, however, so that's a consideration. I'm not against this change but it would require additional time/effort. Many test functions, for example, make use of `Sphere` as a fixture, so the effort required could be substantial)
- The current test suite for the correct direction/orientation of `Sphere` (introduced in #4674) uses terminology that aims to be consistent with the current docstring terminology and internal language used by VTK. So, if these terms/definitions are modified in the docstring (which they should be as it's obviously unclear/misleading at the moment), I'd probably add comments to the test suites where appropriate to make it clear what's being tested, e.g. if the term "latitude 0" is used in a test, is that VTK language for the north pole or PyVista docstring language for the equator?
- There are some undocumented spherical utilities in `utilities.features`. Even these are not self-consistent, with the third function swapping the meaning theta and phi:
    - [transform_vectors_sph_to_cart](https://github.com/pyvista/pyvista/blob/9ab97a6532fec0aea0d30c48b4a1ec1b4eb7daa7/pyvista/core/utilities/features.py#L174)
        - theta : Azimuthal angle
        - phi : Polar angle
    - [grid_from_sph_coords](https://github.com/pyvista/pyvista/blob/9ab97a6532fec0aea0d30c48b4a1ec1b4eb7daa7/pyvista/core/utilities/features.py#L147)
        - theta : Azimuthal angle
        - phi : Polar angle
    - [cartesian_to_spherical](https://github.com/pyvista/pyvista/blob/9ab97a6532fec0aea0d30c48b4a1ec1b4eb7daa7/pyvista/core/utilities/features.py#L212)
        - theta : Polar angle
        - phi : Azimuthal angle
- I agree `SolidSphere` should use the same convention as `Sphere`. I also think the utility functions above should be self-consistent (they currently are not). I am not so sure, however, that the geometric objects need to match the utility functions, due to effort required / deviation from VTK. 
- Perhaps all of these functions should have a `.. note::` or `.. warning::` section, stating "Many conventions exist for spherical coordinates (see <insert wolfram link> for details). Read the parameter definitions carefully and have a look at the examples below to understand the conventions used by this function."

Reply:
I think we can make terminology unambiguous by calling one angle "polar angle" (I think this has only one possible interpretation), and the other "azimuthal angle where 0 degrees points along the `x` axis" or similar (because even with an azimuthal angle you have some freedom of choosing the zero point).

Reply:
> I agree that this choice is confusing, and that a north pole at +90 degrees latitude makes more sense. To redefine this would require deviation from its current alignment with VTK usage, however, so that's a consideration. I'm not against this change but it would require additional time/effort. Many test functions, for example, make use of Sphere as a fixture, so the effort required could be substantial)

My suggestion is to let `theta` be the azimuthal angle running from `0` to `2pi` (or `0` to `360` degrees) and let `phi` be the polar angle running from `0` to `pi` (or `0` to `180` degrees). My guess is that this is how `pyvista` and `vtk` are currently coded and this doesn't require any change to any *code*.

My suggestion was to remove the "latitude 0 degrees" language when referring to the North pole. In my mind it is straight up wrong to say the North pole is at latitude 0 degrees, and if VTK docs use that language then an issue should be opened with them. But I believe this is only a change to documentation, not any code, so I would be surprised if test functions using `Sphere` would need to be modified.

@adeak I agree that "polar angle" and "azimuthal angle" are unambiguous and I think that terminology should be used in the docs. But as much as I dislike the theta = azimuthal and phi = polar angle for the parameter names, I think it is an established convention and should be stuck with. That is, I don't think the parameter names should be changed from `theta` and `phi` to `azimuthal_angle` and `polar_angle`. The latter is just too verbose. Also the ordering of `theta` and `phi` in the function signature should just match whatever `sphere` and `VTK` use, just for consistency.

Reply:
Ok yeah, looking at [the documentation for Sphere](https://docs.pyvista.org/version/stable/api/utilities/_autosummary/pyvista.Sphere.html). It looks like someone was trying to use `latitude` and `longitude` to be a helper to readers to keep straight whether `theta` or `phi` is azimuthal or polar, but in doing so they introduced the badness of referring to the North pole as "zero degrees latitude". I would say the "longitude" and "latitude" names could be preserved in high level documentation saying things like "Pyvista uses a convention where theta represents the azimuthal angle (similar to degrees longitude on the globe) and phi represents the polar angle (similar to degrees latitude on the globe. Note that this is different than a convention chosen in some domains where theta is the polar angle and phi is the azimuthal angle.". But in the API documentation I think longitude and latitude should NOT be used and azimuthal and polar terminology should always be preferred. 

Maybe a geo-sciences person can chime in and indicate that in some cases they use zero degrees latitude to mean the North pole, but I doubt it.

Reply:
> in the API documentation I think longitude and latitude should NOT be used and azimuthal and polar terminology should always be preferred.

I agree this is a big source of the confusion and that the longitude and latitude should be removed from `Sphere`. Polar angle and azimuthal angle is much clearer.

I also agree that it's fine to keep with the current parameter names, i.e. `theta` -> azimuthal and `phi` -> polar angle by convention if there's good documentation. ([cartesian_to_spherical](https://github.com/pyvista/pyvista/blob/9ab97a6532fec0aea0d30c48b4a1ec1b4eb7daa7/pyvista/core/utilities/features.py#L212) does not follow this convention, and is the odd one out in this regards in terms of consistency in PyVista... this can be raised as a separate issue)

Reply:
> @adeak I agree that "polar angle" and "azimuthal angle" are unambiguous and I think that terminology should be used in the docs. But as much as I dislike the theta = azimuthal and phi = polar angle for the parameter names, I think it is an established convention and should be stuck with. That is, I don't think the parameter names should be changed from `theta` and `phi` to `azimuthal_angle` and `polar_angle`. The latter is just too verbose. Also the ordering of `theta` and `phi` in the function signature should just match whatever `sphere` and `VTK` use, just for consistency.

@jagerber48 I never suggested changing the variable names, I've been talking about documentation from the get go 😛

Reply:
I have tried to go through and use `azimuthal` and `polar` in all cases.  I used a modified form of @jagerber48 suggestion in the description.  I think we should avoid the different definitions of theta and phi.  Given the Wolfram link has a whole table for this, I think we are safe as long as the definition is clear here.

```
    PyVista uses a convention where theta represents the azimuthal
    angle (similar to degrees longitude on the globe) and phi
    represents the polar angle (similar to degrees latitude on the
    globe). In contrast to latitude on the globe, here
    phi is 0 degrees at the North Pole and 180 degrees at the South
    Pole.
```